AppInsightsCommon/src/Interfaces/IConfig.ts

```diff
@@ -264,10 +264,10 @@ export interface IConfig {
 export class ConfigurationManager {
     public static getConfig(config: IConfiguration & IConfig, field: string, identifier?: string, defaultValue: number | string | boolean = false): number | string | boolean {
         let configValue;
-        if (!identifier || !config.extensionConfig || !config.extensionConfig[identifier]) {
-            configValue = config[field];
-        } else {
+        if (identifier && config.extensionConfig && config.extensionConfig[identifier] && !CoreUtils.isNullOrUndefined(config.extensionConfig[identifier][field])) {
```

Comment:
was this still buggy?

Reply:
yes need to check for not undefined instead of truthy